plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -255,38 +258,58 @@ function simpleRepeatingInputMixin(mutatorContainer, mutatorInput, inputName, in
             // Count number of inputs.
             const connections = [];
+            const fieldValues = [];
             while (inputBlock && !inputBlock.isInsertionMarker()) {
                 connections.push(inputBlock.valueConnection_);
+                fieldValues.push(inputBlock.fieldValues_);
                 inputBlock = inputBlock.nextConnection && inputBlock.nextConnection.targetBlock();
             }
-            // Disconnect any children that don't belong.
-            for (let i = 0; i < this.inputCount_; i++) {
-                const connection = this.getInput(inputName + i) && this.getInput(inputName + i).connection.targetConnection;
-                if (connection && connections.indexOf(connection) == -1) {
-                    connection.disconnect();
+            // Disconnect any children that don't belong. This is skipped if the provided input is a dummy input
+            if (isProperInput) {
+                for (let i = 0; i < this.inputCount_; i++) {
+                    const connection = this.getInput(inputName + i) && this.getInput(inputName + i).connection.targetConnection;
+                    if (connection && connections.indexOf(connection) == -1) {
+                        connection.disconnect();
+                    }
                 }
             }
             this.inputCount_ = connections.length;
             this.updateShape_();
-            // Reconnect any child blocks.
+            // Reconnect any child blocks and update the field values
             for (let i = 0; i < this.inputCount_; i++) {
-                Blockly.Mutator.reconnect(connections[i], this, inputName + i);
+                if (isProperInput) {
+                    Blockly.Mutator.reconnect(connections[i], this, inputName + i);
+                }
+                if (fieldValues[i]) {
+                    for (let j = 0; j < fieldNames.length; j++) {
+                        let currentField = this.getField(fieldNames[j] + i);
+                        currentField.setValue(fieldValues[i][j] ?? '');
+                    }
+                }
             }
         },
 
         // Keep track of the connected blocks, so that they don't get disconnected whenever an input is added or moved
+        // This also keeps track of the field values
         saveConnections: function(containerBlock) {
             let inputBlock = containerBlock.getInputTargetBlock('STACK');
             let i = 0;
             while (inputBlock) {
-                if (inputBlock.isInsertionMarker()) {
-                    inputBlock = inputBlock.getNextBlock();
-                    continue;
+                if (!inputBlock.isInsertionMarker()) {
+                    const input = this.getInput(inputName + i);
+                    if (input) {
+                        if (isProperInput) {
+                            inputBlock.valueConnection_ = input && input.connection.targetConnection;

```

Comment:
You already check for input presence in one of surrounding conditions, this duplicate check is redundant.
```suggestion
                            inputBlock.valueConnection_ = input.connection.targetConnection;
```